src/server/server.js

```diff
@@ -1,191 +1,33 @@
-
-    var newUser = {};
-    newUser.email = req.body.email.toLowerCase();
-    newUser.username = req.body.username.trim();
-    newUser.password = req.body.password;
-
-    User.create(newUser, function (err, user) {
-        if (err) {
-            req.flash('error', err.message);
-            return res.redirect('back');
-        } else {
-            // Passport exposes a login() function on req (also aliased as logIn())
-            // that can be used to establish a login session. This function is
-            // primarily used when users sign up, during which req.login() can
-            // be invoked to log in the newly registered user.
-            req.login(user, function (err) {
-                if (err) {
-                    req.flash('error', err.message);
-                    return res.redirect('back');
-                }
-                return res.redirect('/auth/account');
-            });
-        }
-    });
-});
-
-app.get('/login', function (req, res, next) {
-    res.render('pages/login', {
-        user: req.user,
-        url: req.url
-    });
-});
-
-app.get('/auth/logout', function (req, res, next) {
-    rpeq.logout();
-    res.redirect('/');
-});
+var PassportConfigurator =
+  require('loopback-component-passport').PassportConfigurator;
 
-// -- Mount static files here--
-// All static middleware should be registered at the end, as all requests
-// passing the static middleware are hitting the file system
-// Example:
-var path = require('path');
-app.use(loopback.static(path.resolve(__dirname, '../client/public')));
-
-// Requests that get this far won't be handled
-// by any middleware. Convert them into a 404 error
-// that will be handled later down the chain.
-//app.use(loopback.urlNotFound());
-
-// The ultimate error handler.
-app.use(loopback.errorHandler());
+var app = module.exports = loopback();
 
-app.start = function () {
-    // start the web server
-    return app.listen(function () {
-        app.emit('started');
-        var baseUrl = app.get('url').replace(/\/$/, '');
-        console.log('%s %s', __dirname, util.inspect(app.get('url')));
-        console.log('Web server listening at: %s', baseUrl);
-        if (app.get('loopback-component-explorer')) {
-            var explorerPath = app.get('loopback-component-explorer').mountPath;
-            console.log('Browse your REST API at %s%s', baseUrl, explorerPath);
-        }
-    });
+app.start = function() {
+  // start the web server
+  return app.listen(function() {
+    app.emit('started');
+    console.log('Web server listening at: %s', app.get('url'));
+    app.emit('started');
+    var baseUrl = app.get('url').replace(/\/$/, '');
+    console.log('Web server listening at: %s', baseUrl);
+    if (app.get('loopback-component-explorer')) {
+      var explorerPath = app.get('loopback-component-explorer').mountPath;
+      console.log('Browse your REST API at %s%s', baseUrl, explorerPath);
+    }
+  });
 };
 
-app.boot = function (cb) {
```

Comment:
Hi sunil, app.boot coding pattern has been used to background a bug inside
loopback fwk.

https://github.com/strongloop/loopback/issues/1547 (chk the code at the
bottom of the page)

So can we pls follow the model I suggested in my original comment? :)
On Jan 24, 2016 8:21 AM, "snlkapil" <notifications@github.com> wrote:

> In src/server/server.js
> <https://github.com/InsBob/master/pull/1#discussion_r50637671>:
>
> >  };
> >
> > -app.boot = function (cb) {
>
> *Thanks for mentioning it [image: :+1:] *
> #1 <https://github.com/InsBob/master/pull/1> would it be ok, if I add
> these lines on test: (if we use Karma or Jasmine for unit/integration test)
>
> beforeEach(function() {
>     loopback = require('loopback');
>     var boot = require('loopback-boot');
>     app = loopback();
>     boot(app, '../server', function (err) {
>       if (err) throw err;
>       console.log('set bootedApp as app');
>     });
> });
>
> *If it doesn't make sense, I will change the code from boot to app.boot.*
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/InsBob/master/pull/1/files#r50637671>.
>